### ISSUE TITLE
Unit tests crash

### ISSUE THREAD

**Issue by janihur** [ballerina-platform/ballerina-lang#22243]
**Description:**
Running a unit test with `ballerina test --all` crashes with `ballerina: Oh no, something really went wrong. Bad. Sad.`

```
jani@baldev:/tmp/mocks$ ballerina clean && ballerina test --all
Compiling source
        tmp/bar:0.1.0
        tmp/foo:0.1.0
        tmp/main:0.1.0
ballerina: Oh no, something really went wrong. Bad. Sad.

There should be a file named "ballerina-internal.log" in the current directory.
If you are able to share with us the code that broke Ballerina then
we would REALLY appreciate if you would report this to us:
go to https://github.com/ballerina-platform/ballerina-lang/issues and
create a bug report with both this log file and the sample code.

We thank you for helping make us better dancers.
jani@baldev:/tmp/mocks$ cat ballerina-internal.log 
[2020-03-26 19:01:58,030] SEVERE {b7a.log.crash} - String index out of range: 9 
java.lang.StringIndexOutOfBoundsException: String index out of range: 9
        at java.lang.String.substring(String.java:1963)
        at org.ballerinalang.testerina.core.TestAnnotationProcessor.formatPackageName(TestAnnotationProcessor.java:357)
        at org.ballerinalang.testerina.core.TestAnnotationProcessor.lambda$process$1(TestAnnotationProcessor.java:168)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.ballerinalang.testerina.core.TestAnnotationProcessor.process(TestAnnotationProcessor.java:149)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.lambda$visit$7(CompilerPluginRunner.java:166)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.notifyProcessors(CompilerPluginRunner.java:282)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.visit(CompilerPluginRunner.java:166)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:73)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.lambda$visit$5(CompilerPluginRunner.java:155)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.visit(CompilerPluginRunner.java:155)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.lambda$visit$3(CompilerPluginRunner.java:142)
        at java.util.ArrayList.forEach(ArrayList.java:1257)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.visit(CompilerPluginRunner.java:140)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:161)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.CompilerPluginRunner.runPlugins(CompilerPluginRunner.java:126)
        at org.wso2.ballerinalang.compiler.CompilerDriver.annotationProcess(CompilerDriver.java:294)
        at org.wso2.ballerinalang.compiler.CompilerDriver.compile(CompilerDriver.java:252)
        at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackageSymbol(CompilerDriver.java:214)
        at org.wso2.ballerinalang.compiler.CompilerDriver.compilePackage(CompilerDriver.java:131)
        at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:166)
        at org.wso2.ballerinalang.compiler.Compiler.compilePackages(Compiler.java:137)
        at org.ballerinalang.packerina.task.CompileTask.execute(CompileTask.java:65)
        at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
        at org.ballerinalang.packerina.cmd.TestCommand.execute(TestCommand.java:373)
        at java.util.Optional.ifPresent(Optional.java:159)
        at org.ballerinalang.tool.Main.main(Main.java:57)
 
jani@baldev:/tmp/mocks$ 
```

**Steps to reproduce:**

This is the full project code:

```
jani@baldev:/tmp/mocks$ tree
.
├── Ballerina.toml
└── src
    ├── bar
    │   ├── main.bal
    │   └── tests
    │       └── main_test.bal
    ├── foo
    │   ├── main.bal
    │   └── tests
    │       └── main_test.bal
    └── main
        ├── main.bal
        └── tests
            └── main_test.bal

7 directories, 7 files
jani@baldev:/tmp/mocks$ cat src/bar/main.bal 
public function barFun() returns string {
    return "barFun()";
}
jani@baldev:/tmp/mocks$ cat src/bar/tests/main_test.bal 
import ballerina/test;

@test:Config {}
function test1() {
    test:assertEquals(barFun(), "barFun()");
}
jani@baldev:/tmp/mocks$ cat src/foo/main.bal 
public function fooFun() returns string {
    return "fooFun()";
}
jani@baldev:/tmp/mocks$ cat src/foo/tests/main_test.bal 
import ballerina/test;

@test:Config {}
function test1() {
    test:assertEquals(fooFun(), "fooFun()");
}
jani@baldev:/tmp/mocks$ cat src/main/main.bal 
import ballerina/io;

import tmp/bar;
import tmp/foo;

function mainFun() returns string[] {
    return [bar:barFun(), foo:fooFun()];
}

public function main() {
    io:println(mainFun());
}
jani@baldev:/tmp/mocks$ cat src/main/tests/main_test.bal 
import ballerina/test;

@test:Config {}
function test1() {
    test:assertEquals(mainFun(), <string[]>["barFun()", "fooFun()"]);
}

@test:Mock {
    moduleName: "bar",
    functionName: "barFun"
}
function mockBarFun() returns string {
    return "barMock!";
}

@test:Config {}
function test2() {
    test:assertEquals(mainFun(), <string[]>["barMock!", "fooFun()"]);
}
jani@baldev:/tmp/mocks$ 
```

**Affected Versions:**

```
jani@baldev:/tmp/mocks$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 19.04
Release:        19.04
Codename:       disco
jani@baldev:/tmp/mocks$ ballerina --version
jBallerina 1.2.0
Language specification 2020R1
Ballerina tool 0.8.0
jani@baldev:/tmp/mocks$
```

**Comment by Ibaqu** [comment]
Fix is handled in the following PR :

https://github.com/ballerina-platform/ballerina-lang/pull/22348

**Comment by Ibaqu** [comment]
Fix is available in Ballerina ver 1.2.1 hence closing
